device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -219,14 +220,22 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
     m_copy.setup(seeds_buffer);
     m_copy.setup(navigation_buffer);
 
+    const unsigned int n_seeds = m_copy.get_size(seeds_buffer);
+
     // Prepare input parameters with seeds
-    bound_track_parameters_collection_types::buffer in_params_buffer(
-        m_copy.get_size(seeds_buffer), m_mr.main);
+    bound_track_parameters_collection_types::buffer in_params_buffer(n_seeds,
+                                                                     m_mr.main);
     bound_track_parameters_collection_types::device in_params(in_params_buffer);
-    bound_track_parameters_collection_types::device seeds(seeds_buffer);
+    bound_track_parameters_collection_types::const_device seeds(seeds_buffer);
     thrust::copy(thrust::cuda::par.on(stream), seeds.begin(), seeds.end(),
                  in_params.begin());
 
+    // Number of tracks per seed
+    vecmem::data::vector_buffer<unsigned int> n_tracks_per_seed_buffer(
+        n_seeds, m_mr.main);
+    vecmem::device_vector<unsigned int> n_tracks_per_seed_device(
+        n_tracks_per_seed_buffer);
```

Comment:
I'd prefer to not use a `vecmem::device_vector` here. As it could easily result in some unsafe function calls later on. :thinking:

I'd rather use `n_tracks_per_seed_buffer.ptr()` and `n_tracks_per_seed_buffer.ptr() + n_tracks_per_seed_buffer.size()` as the parameters of `thrust::fill`.

Or, now that I think about it, since you're "just" zeroing out the vector, let's not use `thrust::fill` at all. You should instead write:

```c++
m_copy.memset(n_tracks_per_seed_buffer, 0)->ignore();
```

When using `vecmem::cuda::async_copy`, this will be an asynchronous operation out of the box.

Reply:
Thanks for the useful syntax. Fixed now